src/unzip/mod.rs

```diff
@@ -527,6 +551,25 @@ mod tests {
         zip.finish().unwrap();
     }
 
+    fn create_encrypted_zip(w: impl Write + Seek, include_a_txt: bool) {
+        let mut zip = ZipWriter::new(w);
+
+        zip.add_directory("test/", Default::default()).unwrap();
+        let options = FileOptions::default()
+            .compression_method(zip::CompressionMethod::Stored)
+            .unix_permissions(0o755)
+            .with_deprecated_encryption("1Password".as_ref());
+        if include_a_txt {
```

Comment:
Please could you remove the duplication here related to the contents of the zip.

Reply:
thanks for this. 

not sure if it's what you had in mind, but i've pushed a re-factor

Reply:
Nope, it's better than what I had in mind :)